### PR TITLE
K8SPXC-260 Add Openshift securitycontextcontstraints support

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| storage | passed |
| init-deploy | passed |
| big-data | passed |
| scaling | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| limits | passed |
| demand-backup | passed |
| upgrade | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| security-context | passed |
| monitoring | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| scheduled-backup | failed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| scaling | passed |
| storage | failed |
| recreate | passed |
| big-data | passed |
| scaling-proxysql | passed |
| limits | passed |
| demand-backup | failed |
| upgrade | passed |
| self-healing | failed |
| monitoring | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| scheduled-backup | failed |
| self-healing-advanced | passed |
| affinity | passed |
| one-pod | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| init-deploy | passed |
| scaling | passed |
| recreate | passed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| limits | passed |
| monitoring | passed |
| demand-backup | passed |
| upgrade | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| affinity | passed |
| operator-self-healing | failed |
| security-context | failed |
| scheduled-backup | failed |
| one-pod | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| recreate | passed |
| storage | passed |
| scaling | passed |
| big-data | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| limits | passed |
| demand-backup | failed |
| upgrade | failed |
| self-healing-advanced | passed |
| monitoring | passed |
| upgrade-consistency | passed |
| security-context | failed |
| monitoring-2-0 | failed |
| operator-self-healing | failed |
| scheduled-backup | failed |
| affinity | failed |
| one-pod | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

File: pkg/controller/pxc/upgrade.go [hunk 1]
```diff
@@ -42,6 +42,15 @@ func (r *ReconcilePerconaXtraDBCluster) updatePod(sfs api.StatefulApp, podSpec *
 		currentSet.Spec.Template.Annotations = make(map[string]string)
 	}
 	if cr.CompareVersionWith("1.1.0") >= 0 {
+		if r.serverVersion.Platform == api.PlatformOpenshift && cr.Spec.ProxySQL.ContainerSecurityContext != nil {
```
Comment (delgod) [inline pkg/controller/pxc/upgrade.go:45]: should be tunable via annotations in CR

File: pkg/controller/pxc/upgrade.go [hunk 2]
```diff
@@ -111,6 +120,11 @@ func (r *ReconcilePerconaXtraDBCluster) updatePod(sfs api.StatefulApp, podSpec *
 	}
 
 	currentSet.Spec.Template.Spec.Containers = newContainers
+	if r.serverVersion.Platform == api.PlatformOpenshift && cr.Spec.ProxySQL.ContainerSecurityContext != nil {
```
Comment (delgod) [inline pkg/controller/pxc/upgrade.go:123]: should be tunable via CSC in CR

File: pkg/controller/pxc/controller.go [hunk 3]
```diff
@@ -348,6 +360,17 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 			nodeSet.Spec.Template.Annotations = make(map[string]string)
 		}
 		if cr.CompareVersionWith("1.1.0") >= 0 {
+			if r.serverVersion.Platform == api.PlatformOpenshift && cr.Spec.ProxySQL.ContainerSecurityContext != nil {
+				if *cr.Spec.ProxySQL.ContainerSecurityContext.Privileged {
```
Comment (delgod) [inline pkg/controller/pxc/controller.go:364]: all should be tunable via annotations, PodSecurityContext in CR
maybe needed to add an option for proxySet.Spec.Template.Spec.ServiceAccountName

File: pkg/controller/pxc/controller.go [hunk 4]
```diff
@@ -273,6 +273,18 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 		nodeSet.Spec.Template.Annotations = make(map[string]string)
 	}
 	if cr.CompareVersionWith("1.1.0") >= 0 {
+		if r.serverVersion.Platform == api.PlatformOpenshift && cr.Spec.ProxySQL.ContainerSecurityContext != nil {
+			if *cr.Spec.ProxySQL.ContainerSecurityContext.Privileged {
+				nodeSet.Spec.Template.Annotations["openshift.io/scc"] = "privileged"
+				nodeSet.Spec.Template.Spec.Containers[0].SecurityContext = nil
+				nodeSet.Annotations = map[string]string{"openshift.io/scc": "privileged"}
+				nodeSet.Spec.Template.Spec.ServiceAccountName = "percona-operations"
```
Comment (delgod) [inline pkg/controller/pxc/controller.go:281]: all should be tunable via annotations, PodSecurityContext in CR
maybe needed to add an option for nodeSet.Spec.Template.Spec.ServiceAccountName

**Comment by delgod** [comment]
should be reworked after https://github.com/percona/percona-xtradb-cluster-operator/pull/417 merge

**Comment by delgod** [comment]
now, everything can be passed via CR, ServiceAccountName was added in https://github.com/percona/percona-xtradb-cluster-operator/pull/417

so we can reach the goal just via changing values in tests CR

**Pull request closed without merging**